### PR TITLE
Chain favourite

### PR DESCRIPTION
**Pull request by XiaoYhun**

### CODE REVIEW

File: src/pages/TrueSightV2/components/TokenFilter/index.tsx [hunk 1]
```diff
@@ -246,9 +246,9 @@ export default function TokenFilter({
               <StyledSelect
                 value={filter[queryKey]}
                 key={queryKey}
-                activeRender={item => <ActiveSelectItem name={displayName} label={item?.label} />}
+                activeRender={(item: any) => <ActiveSelectItem name={displayName} label={item?.label} />}
```
Comment (nguyenhoaidanh) [inline src/pages/TrueSightV2/components/TokenFilter/index.tsx:249]: I dont see any warning/error on this.
image

File: src/components/Header/web3/NetworkModal/index.tsx [hunk 2]
```diff
@@ -1,22 +1,71 @@
...
+  return (
+    <div
+      ref={ref}
+      onDragOverCapture={() => setIsDraggingOver(true)}
+      style={
+        isDraggingOver
+          ? {
+              backgroundColor: 'red',
+            }
+          : {} //todo
```
Comment (nguyenhoaidanh) [inline src/components/Header/web3/NetworkModal/index.tsx:58]: todo

File: src/components/Header/web3/NetworkModal/index.tsx [hunk 3]
```diff
@@ -33,40 +82,188 @@ export default function NetworkModal({
...
+  const { supportedChains } = useChainsConfig()
+
+  const handleDrop = (chainId: string, dropId: string) => {
+    if (dropId === FAVORITE_DROPZONE_ID) {
+      const chainInfo = supportedChains.find(item => item.chainId.toString() === chainId)
+      if (chainInfo && !favoriteChains.includes(chainInfo)) {
+        saveFavoriteChains([...favoriteChains, chainInfo.chainId.toString()])
+      }
+    }
+    if (dropId === CHAINS_DROPZONE_ID) {
```
Comment (nguyenhoaidanh) [inline src/components/Header/web3/NetworkModal/index.tsx:108]: if(dropId === CHAINS_DROPZONE_ID&&(dropId === CHAINS_DROPZONE_ID))...

File: .github/workflows/pr.yaml [hunk 4]
```diff
@@ -114,7 +114,7 @@ jobs:
           VITE_TAG: ${{ needs.prepare.outputs.image_tag }}
           CURRENT_BRANCH: ${{ needs.prepare.outputs.current_branch }}
           NODE_OPTIONS: '--max_old_space_size=4096'
-        run: yarn build
+        run: yarn build-dev
```
Comment (nguyenhoaidanh) [inline .github/workflows/pr.yaml:117]: revert

File: src/components/Header/web3/NetworkModal/DraggableNetworkButton.tsx [hunk 5]
```diff
@@ -0,0 +1,316 @@
...
+        layoutId={networkInfo.chainId.toString() + networkInfo.route}
+        selected={selected}
+        animate={animateControls}
+        transition={{ type: 'spring', damping: 50, stiffness: 1000 }}
+        variants={variants}
+        style={{ boxShadow: '0 0px 0px #00000060' }}
+        onClick={() => !selected && !isDragging && handleChainSelect()}
+        $disabled={disabled}
+      >
+        <img src={icon} alt="Switch Network" style={{ height: '20px', width: '20px' }} />
```
Comment (namgold) [inline src/components/Header/web3/NetworkModal/DraggableNetworkButton.tsx:266]:
```suggestion
        <img src={icon} alt="Switch Network" style={{ maxHeight: '20px', maxWidth: '20px' }} />
```

File: src/components/Header/web3/NetworkModal/index.tsx (hunk 3 above)
Comment (XiaoYhun) [reply src/components/Header/web3/NetworkModal/index.tsx:85]: I don't get this

File: src/components/Header/web3/NetworkModal/DraggableNetworkButton.tsx (hunk 5 above)
Comment (XiaoYhun) [reply src/components/Header/web3/NetworkModal/DraggableNetworkButton.tsx:266]: copy from legacy code, I think unnecessary to update

File: src/components/Header/web3/NetworkModal/index.tsx (hunk 3 above)
Comment (nguyenhoaidanh) [reply src/components/Header/web3/NetworkModal/index.tsx:85]: if(dropId === CHAINS_DROPZONE_ID &&  favoriteChains.some(fChainId => fChainId === chainId) ){}

**Comment by nguyenhoaidanh** [comment]
lgtm, wait for nanogold continue his review

File: src/components/Header/web3/NetworkModal/index.tsx [hunk 6]
```diff
@@ -33,40 +59,188 @@ export default function NetworkModal({
...
-        />
+          </ButtonAction>
+        </RowBetween>
+
+        <Column marginTop="16px" gap="8px">
+          <Row gap="12px">
+            <Text fontSize="10px" lineHeight="24px" color={theme.subText} flexShrink={0}>
+              <Trans>Favorite Chain(s)</Trans>
+            </Text>
+            <hr style={{ border: '0 0 1px 0', borderColor: theme.border, width: '100%' }} />
```
Comment (namgold) [inline src/components/Header/web3/NetworkModal/index.tsx:162]: `border: '0 0 1px 0'` this is not working. Can you test again

File: src/components/Header/web3/NetworkModal/index.tsx [hunk 7]
```diff
@@ -33,40 +59,188 @@ export default function NetworkModal({
...
+                  })}
+              </NetworkList>
+            )}
+          </div>
+
+          <Row gap="12px">
+            <Text fontSize="10px" lineHeight="24px" color={theme.subText} flexShrink={0}>
+              <Trans>Chain List</Trans>
+            </Text>
+            <hr style={{ border: '0 0 1px 0', borderColor: theme.border, width: '100%' }} />
```
Comment (namgold) [inline src/components/Header/web3/NetworkModal/index.tsx:222]: `border: '0 0 1px 0'` this is not working. Can you test again

Comment (namgold) [reply src/components/Header/web3/NetworkModal/index.tsx:222]:
```suggestion
            <hr style={{ borderWidth: '0 0 1px 0', borderColor: theme.border, width: '100%' }} />
```

File: src/components/Header/web3/NetworkModal/index.tsx (hunk 6 above)
Comment (namgold) [reply src/components/Header/web3/NetworkModal/index.tsx:162]:
```suggestion
            <hr style={{ borderWidth: '0 0 1px 0', borderColor: theme.border, width: '100%' }} />
```

Comment (XiaoYhun) [reply src/components/Header/web3/NetworkModal/index.tsx:162]: yeah right, but it showed perfectly on UI :)))

**Pull request merged**